ihontaryk/homework_8/warrior.py

```diff
@@ -0,0 +1,58 @@
+from logging import getLogger
+import logging
+
+
+class Player:
+    def __init__(self, health=50, attack=5):
+        self.health = health
+        self.attack = attack
+
+    @property
+    def is_alive(self):
+        return self.health > 0
+
+    def hit(self, player: 'Player'):
+        player.lose_health(self.attack)
+
+    def lose_health(self, damage: int):
+        self.health = max(0, self.health - damage)
+
+
+class Warrior(Player):
+    def __init__(self):
+        super().__init__()
+
+
+class Knight(Player):
+    def __init__(self):
+        super().__init__()
+        self.attack = 7
+
+
+def fight(player1: Player, player2: Player) -> bool:
+    round = 1
+    logger = getLogger(__name__)
+    logging.basicConfig(filename='warrior.log', encoding='utf-8', level=logging.INFO)
```

Comment:
logging configuration should be moved out of here, somewhere to the top, e.g. to 
tests/__init__.py